src/__experimental__/components/date/date.component.js

```diff
@@ -0,0 +1,259 @@
+import React from 'react';
+import I18n from 'i18n-js';
+import PropTypes from 'prop-types';
+import Events from '../../../utils/helpers/events';
+import DateHelper from '../../../utils/helpers/date';
+import DateValidator from '../../../utils/validations/date';
+import tagComponent from '../../../utils/helpers/tags';
+import DatePicker from './date-picker.component';
+import InputDecoratorBridge from '../input-decorator-bridge';
+import StyledDateInput from './date.style';
+import Textbox from '../textbox';
+
+const isoDateFormat = 'YYYY-MM-DD';
+const today = DateHelper.todayFormatted(isoDateFormat);
+
+/**
+ * A Date widget.
+ *
+ * == How to use a Date in a component:
+ *
+ * In your file
+ *
+ *   import Date from 'carbon-react/lib/components/Date';
+ *
+ * To render the Date:
+ *
+ *   <Date name="myDate" />
+ *
+ * @class Date
+ * @constructor
+ * @decorators {Input,InputIcon,InputLabel,InputValidation}
+ */
+class Date extends React.Component {
+  static propTypes = {
+    ...Textbox.propTypes,
+    /** Automatically focus on component mount */
+    autoFocus: PropTypes.bool,
+    /** Used to provide additional validations on composed components */
+    internalValidations: PropTypes.array,
+    /** Minimum possible date YYYY-MM-DD */
+    minDate: PropTypes.string,
+    /** Maximum possible date YYYY-MM-DD */
+    maxDate: PropTypes.string,
+    /** Specify a callback triggered on blur */
+    onBlur: PropTypes.func,
+    /** Specify a callback triggered on change */
+    onChange: PropTypes.func,
+    /** Specify a callback triggered on focus */
+    onFocus: PropTypes.func,
+    /** The current date */
+    value: PropTypes.string
+  };
+
+  static defaultProps = {
+    value: today,
+    internalValidations: [new DateValidator()]
+  };
+
+  // Stores the document - allows us to override it different contexts, such as when running tests.
+  _document = document;
+
+  bridge = React.createRef() // this is a reference to the input decorator bridge component
+
+  blurBlocked = false; // stops the blur callback from triggering (closing the list) when we don't want it to
+
+  state = {
+    isDatePickerOpen: false,
+    selectedDate: null,
+    visibleValue: this.formatVisibleValue(this.props.value)
+  };
+
+  componentDidMount() {
+    if (this.props.autoFocus) {
+      this.blockFocus = true;
+      this.input.focus();
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.blurBlocked && this.hasDatePickerValueChanged(prevProps)) {
+      this.unblockBlur();
+      this.handleBlur(); // TODO validate
+    }
+  }
+
+  hasDatePickerValueChanged = (prevProps) => {
```

Comment:
Should be `datePickerValueHasChanged`

Reply:
Is there a reason why it should be changed? That method could be treated as a question which makes more sense in my opinion, as the returned value is a boolean.